src/main/java/org/jabref/gui/externalfiles/DownloadFullTextAction.java

```diff
@@ -102,23 +106,15 @@ protected Map<BibEntry, Optional<URL>> call() {
                 Localization.lang("Looking for full text document..."),
                 findFullTextsTask);
 
-        Globals.TASK_EXECUTOR.execute(findFullTextsTask);
+        taskExecutor.execute(findFullTextsTask);
     }
 
     private void downloadFullTexts(Map<BibEntry, Optional<URL>> downloads, BibDatabaseContext databaseContext) {
         for (Map.Entry<BibEntry, Optional<URL>> download : downloads.entrySet()) {
             BibEntry entry = download.getKey();
             Optional<URL> result = download.getValue();
             if (result.isPresent()) {
-                Optional<Path> dir = databaseContext.getFirstExistingFileDir(preferences.getFilePreferences());
```

Comment:
why did you remove this?

Reply:
It was not used anymore since a while, but not removed before.
LinkedFileViewModel::download determines itself where to download to.